azure_pipeline-global.yaml

```diff
@@ -0,0 +1,335 @@
+            thisbuild: $(Build.BuildId)
+            pipelinedefinition: $(System.DefinitionId)
+          inputs:
+            targetType: 'inline'
+            script: | 
+              #!/bin/bash
+              set -e
+              organization="hmcts"
+              project="CNP"
+
+              echo "This is build $thisbuild"
+              IFS=$'\n'
+              JSON_DATA=($(curl -s -u :$(azure-devops-token) --request GET "https://dev.azure.com/$organization/$project/_apis/build/builds?api-version=5.1&definitions=$pipelinedefinition" -H "Content-Type: application/json" | jq  '.value[] | .status + (.id|tostring)' | sort -u | grep inProgress))
+              buildnumber=(${JSON_DATA//[!0-9]/})
+
+
+              if [ $thisbuild -eq $buildnumber ] 
+              then
+
+              echo "No other builds are in Progress"
+
+              else
+
+              while [[ $thisbuild -ne $buildnumber_inprogress ]]
+              do 
+
+              IFS=$'\n'
+              JSON_CHECK=($(curl -s -u :$(azure-devops-token) --request GET "https://dev.azure.com/$organization/$project/_apis/build/builds?api-version=5.1&definitions=$pipelinedefinition" -H "Content-Type: application/json" | jq  '.value[] | .status + (.id|tostring)' | sort -u | grep inProgress))
+              buildnumberrunning=(${JSON_CHECK//[!0-9]/})
+              echo "Build number: $buildnumberrunning is in progress"
+
+              buildnumber_inprogress=$buildnumberrunning
+              sleep 30
+
+              done
+
+              fi
+              
+              echo "Starting $thisbuild"
+
+  - stage: GetArtifacts
+    dependsOn: Precheck
+    jobs:
+      - job: Validate
+        pool: ${{ variables.agentPool }}
+        steps:
+          - task: TerraformInstaller@0
+            inputs:
+              terraformVersion: ${{ variables.terraformVersion }}
+          - task: PowerShell@2
+            displayName: Run tests
+            inputs:
+              targetType: 'inline'
+              script: |
+                Install-Module -Name Pester -MaximumVersion 4.99.99 -Force -Verbose -Scope CurrentUser
+                Invoke-Pester ./ci -OutputFormat NUnitXml -OutputFile ./TEST-CI.xml -EnableExit
+              pwsh: true
+          - task: PublishTestResults@2
+            displayName: 'Publish Test Results **/TEST-*.xml'
+            inputs:
+              testResultsFormat: NUnit
+              failTaskOnFailedTests: true
+            condition: always()
+
+      - template: pipeline-templates/terraform-plan-global.yaml
+        parameters:
+          environment: 'sbox'
+          component: 'global'
+          service_connection: 'dcd-cftapps-sbox'
+          storage_account_rg: 'core-infra-sbox-rg'
+          storage_account_name: 'cftappssbox'
+          build: $(Build.BuildNumber)
+          agentPool: ${{ variables.agentPool }}
+          terraformVersion: ${{ variables.terraformVersion }}
+
+      - template: pipeline-templates/terraform-plan-global.yaml
```

Comment:
would be nice to template this https://github.com/hmcts/rdo-sendgrid/blob/master/azure-pipelines.yml#L35

Reply:
yeah, this would be really good to do.

Reply:
See latest PR - I have went for a templated approach